### DISCUSSION TITLE
How to specify the namespace of the service and route

### DISCUSSION THREAD

**Discussion by vbudilov** [apache/apisix#6036] [General]
I’m having a hard time setting up the routes…after I install the gateway using these instructions in its own namespace, I install the routes in the namespace where my services live. How do I connect the two? How do I tell the gateway which namespace to look in? I thought I had to modify the ingress-controller.config.apisix.serviceNamespace variable by specifying the service namespace but that didn’t work.

**Comment by juzhiyuan** [comment]
Hi, I'm not sure if this issue provides necessary information, cc @tao12345666333 to have a look :)

**Comment by tao12345666333** [comment]
By default ingress controller will watch ApisixRoute resources in all namespaces and convert them to Gateway.

**Comment by tokers** [comment]
The backend service should be resided in the same namespace of the `ApisixRoute` object, and just like @tao12345666333 said, by default Apache APISIX Ingress Controller watches all namespaces, so I think you may don't have to modify the `serviceNamespace` option.
